src/jmclient/wallet.py

```diff
@@ -2116,7 +2110,7 @@ def _get_supported_address_types(cls):
 
     def get_script_from_path(self, path):
         if not self._is_my_bip32_path(path):
-            raise WalletError("unable to get script for unknown key path")
+            return super().get_script_from_path(path)
```

Comment:
As far as I understand, I do like this logic better.

Starting from a current hierarchy of (ImportWallet), (BIP32Wallet), (BaseWallet), you're making it so that we always try to get the script as part of the BIP32 hierarchy for this wallet (we always have that BIP32 structure); if we fail, we fall back on a generic "type discovery" routine (get_key_from_path), which then gives us a 'cryptoengine' (never liked that name but whatever) that will be able to output a script for a key. This process can of course fail, but we just see the failure occur in a different place than before, and more importantly, now if we have new Mixins (or superclasses) and support new script types/engines it should naturally slot in, I think.

Side note: I have always strongly disliked the idea of importing keys in the wallet (even since literally the first few weeks of the project), so I have paid very little attention to the import functions, never using them myself. I did have to spend many painful hours figuring out details about private key formats back in the day though, to ensure nothing went catastrophically wrong. All this to say: I would really much prefer to remove any such functionality, but I know that people have used it quite a bit, so I'm rather torn on the subject.

Reply:
Good summary, very helpful, thanks, on the connections between these calls, but:

`_get_pubkey_from_path` doesn't exist though? Either before or after this patch? Or I'm being very dumb here :) There is only _get_key_from_path which is used for `get_script_from_path` and `get_address_from_path`. I could certainly see why it might be useful though!